hkl/configuration.py

```diff
@@ -158,33 +193,34 @@ def validate_config_dict(self, config):
         from .util import libhkl
 
         diffractometer = self.diffractometer
-        assert isinstance(diffractometer, Diffractometer)
-        assert isinstance(config, dict)
 
-        for k, types in EXPECTED_CONFIGURATION_KEYS_TYPES.items():
-            assert k in config
-            assert isinstance(config[k], types)
+        _check_type(self.diffractometer, Diffractometer, "diffractometer should be 'Diffractometer' or subclass.")
+        _check_type(config, dict, "config")
 
-        calc = diffractometer.calc  # shortcut
-        assert config["canonical_axes"] == self.canonical_axes_names
-        assert config["engine"] == calc.engine.name
-        assert config["geometry"] == calc._geometry.name_get()
-        assert config["library"].split()[0] == libhkl.__name__
-        assert config["reciprocal_axes"] == self.reciprocal_axes_names
+        for k, types in REQUIRED_CONFIGURATION_KEYS_TYPES.items():
+            _check_key(k, config, "missing key")
+            _check_type(config[k], types, f"Wrong type for parameter, {k}")
+
+        calc = diffractometer.calc
+        _check_value(
+            config["canonical_axes"],
+            self.canonical_axes_names,
+            "canonical_axes",
+        )
+        _check_value(config["engine"], calc.engine.name, "engine")
+        _check_value(config["geometry"], calc._geometry.name_get(), "geometry")
+        _check_value(config["library"], libhkl.__name__, "library")
+        _check_value(config["reciprocal_axes"], self.reciprocal_axes_names, "reciprocal_axes")
```

Comment:
Same comment as above...

```suggestion
        _check_value(set(config["reciprocal_axes"]), set(self.reciprocal_axes_names), "reciprocal_axes")
```

Reply:
superseded by changes from branch